lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,650 @@
+                        internal_log_rate_limit = true
+                    );
+                }
+            });
+            MessageState::new(total_chunks, timeout_handle)
+        });
+
+        if message_state.total_chunks != total_chunks {
+            warn!(message_id = "Received a chunk with a different total chunks than the original. Ignoring it.",
+                message_id = message_id,
+                original_total_chunks = message_state.total_chunks,
+                received_total_chunks = total_chunks,
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+
+        if message_state.is_chunk_present(sequence_number) {
+            // TODO: add a PR comment asking for info or warn
+            info!(
+                message = "Received a duplicate chunk. Ignoring it.",
+                message_id = message_id,
+                sequence_number = sequence_number,
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+
+        message_state.add_chunk(sequence_number, chunk);
+
+        if let Some(message) = message_state.retrieve_message() {
+            state_lock.remove(&message_id);
+            Ok(Some(message))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Decode a GELF message that may be chunked or not. The source bytes are expected to be
+    /// datagram-based (or message-based), so it must not contain multiple GELF messages
+    /// delimited by '\0', such as it would be in a stream-based protocol.
+    pub fn decode_message(&mut self, mut src: Bytes) -> Result<Option<Bytes>, BoxedFramingError> {
+        let magic = src.get(0..2);
+        if magic.is_some_and(|magic| magic == GELF_MAGIC) {
+            src.advance(2);
+            let frame = self.decode_chunk(src);
+            return frame;
+        } else {
+            return Ok(Some(src));
+        }
+    }
+}
+
+impl Default for ChunkedGelfDecoder {
+    fn default() -> Self {
+        Self::new(DEFAULT_TIMEOUT_MILLIS, DEFAULT_PENDING_MESSAGES_LIMIT)
+    }
+}
+
+impl Decoder for ChunkedGelfDecoder {
+    type Item = Bytes;
+
+    type Error = BoxedFramingError;
+
+    fn decode(&mut self, src: &mut bytes::BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        if src.is_empty() {
+            return Ok(None);
+        }
+
+        // TODO: add a PR comment here stating that this will never call the decode_message since
+        // the bytes decoder will always return a Ok(None) in this method, but leaving this
+        // here for consistency. Would be better to add a unreachable/panic here if the inner decoder returns
+        // the Some variant?
+        self.bytes_decoder
+            .decode(src)?
+            .and_then(|frame| self.decode_message(frame).transpose())
+            .transpose()
+    }
+    fn decode_eof(&mut self, buf: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
```

Comment:
As stated before in the `ChunkedGelfDecoder` struct declaration, we will mainly use the `decode_eof` function for decoding, as we have to read all the bytes from the buffer for the chunked gelf decoding